lampe/inference.py

```diff
@@ -524,14 +540,183 @@ def forward(self, theta: Tensor, x: Tensor) -> Tensor:
 
         theta_prime = torch.roll(theta, 1, dims=0)
 
-        b = self.mask_dist.sample(theta.shape[:-1])
+        b = self.mask_dist.sample(theta.shape[:1])
         theta = torch.where(b, theta, theta_prime)
 
         log_prob = self.estimator(theta, x, b)
 
         return -log_prob.mean()
 
 
+class NSE(nn.Module):
+    r"""Creates a neural score estimation (NSE) regression network.
+
+    The principle of neural score estimation is to train a regression network
+    :math:`s_\phi(\theta, x, t)` to approximate the score of a variance preserving
+    stochastic diffusion process
```

Comment:
Using the "sub-VP SDE" name requires "VP SDE" to be introduced first and maybe a comparison, which I think would be confusing.

Besides, I don't like the name "VP SDE" for the DDPM-style diffusion process as it is not the only variance preserving one. In fact the "sub-VP SDE" is also variance preserving, it just has a lower variance than the "VP SDE".

Reply:
Given that Song 2021 introduces at least two different variance-preserving processes, don't you think it would make it less confusing to tell explicitly the users which variant this class implements? To me "sub-VP SDEs" could be used as is if we just refer to the paper for further details. This name should be known to researchers interested in diffusion-based generative models.